Sources/MongoMobile/MongoMobile.swift

```diff
@@ -52,46 +82,55 @@ private func getStatusExplanation(_ status: OpaquePointer?) -> String {
 /// A class containing static methods for working with MongoMobile.
 public class MongoMobile {
     private static var libraryInstance: OpaquePointer?
+    /// Cache embedded instances for cleanup
     private static var embeddedInstances = [String: OpaquePointer]()
-
+    /// Cache embedded clients for cleanup
+    private static var embeddedClients = [WeakRef<MongoClient>]()
+    
     /**
      * Perform required operations to initialize the embedded server.
      */
     public static func initialize() throws {
         MongoSwift.initialize()
-
+        
         let status = mongo_embedded_v1_status_create()
         var initParams = mongo_embedded_v1_init_params()
         initParams.log_callback = mongo_mobile_log_callback
         initParams.log_flags = UInt64(MONGO_EMBEDDED_V1_LOG_CALLBACK.rawValue)
-
+        
         guard let instance = mongo_embedded_v1_lib_init(&initParams, status) else {
             throw MongoMobileError.invalidInstance(message: getStatusExplanation(status))
         }
-
+        
         libraryInstance = instance
     }
-
+    
     /**
      * Perform required operations to clean up the embedded server.
      */
     public static func close() throws {
+        self.embeddedClients.forEach { ref in
+            ref.reference?.close()
+        }
+        
         let status = mongo_embedded_v1_status_create()
         for (_, instance) in embeddedInstances {
             let result = mongo_embedded_v1_instance_destroy(instance, status)
-            if result == MONGO_EMBEDDED_V1_ERROR_EXCEPTION.rawValue {
-                throw MongoMobileError.instanceDropError(message: getStatusExplanation(status))
+            if result != 0 {
+                throw MongoEmbeddedV1Error.init(result,
```

Comment:
nit: do we need to explicitly say `.init`?

Reply:
Force of habit. I think it looks nicer, but I won't be pedantic 😄